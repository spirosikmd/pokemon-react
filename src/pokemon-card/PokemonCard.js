import React, { PureComponent } from 'react';
import { Card, Image } from 'semantic-ui-react';
import pokeballImage from './pokeball.png';

const styles = {
  myPokemon: {
    opacity: 0.3,
  },
};

class PokemonCard extends PureComponent {
  handlePokemonClick(pokemon) {
    this.props.onPokemonClick(pokemon);
  }

  handleCatchClick(pokemonName) {
    this.props.onCatchClick(pokemonName);
  }

  getStyle(isMyPokemon) {
    let style = {
      boxShadow: '0 1px 3px 0 #D4D4D5, 0 0 0 1px #D4D4D5',
      borderRadius: '50%',
    };
    if (isMyPokemon) {
      style = Object.assign({}, styles.myPokemon, style);
    }
    return style;
  }

  render() {
    const pokemon = this.props.pokemon;

    return (
      <Card link onClick={() => this.handlePokemonClick(pokemon)}>
        <Image
          verticalAlign="middle"
          centered
          src={pokemon.image}
          style={{ margin: '20px auto' }}
          height={100}
        />
        <Card.Content>
          <Card.Header>
            {pokemon.name}
          </Card.Header>
          <Card.Description>
            {pokemon.classification}
          </Card.Description>
        </Card.Content>
        <Card.Content extra>
          <Image
            centered
            size="mini"
            src={pokeballImage}
            style={this.getStyle(this.props.isMyPokemon)}
            onClick={event => {
              event.stopPropagation();
              this.handleCatchClick(pokemon.name);
            }}
          />
        </Card.Content>
      </Card>
    );
  }
}

export default PokemonCard;
