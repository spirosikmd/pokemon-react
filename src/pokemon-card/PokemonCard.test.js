import React from 'react';
import renderer from 'react-test-renderer';
import PokemonCard from './PokemonCard';

const pokemon = {
  name: 'pikachu',
};

const mockHandleOnPokemonClick = jest.fn();
const mockHandleOnCatchClick = jest.fn();

test('PokemonRow renders pokemon name catch button and handle data change', () => {
  const component = renderer.create(
    <PokemonCard
      pokemon={pokemon}
      onPokemonClick={mockHandleOnPokemonClick}
      onCatchClick={mockHandleOnCatchClick}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const name = tree.children[0];
  name.props.onClick();
  expect(mockHandleOnPokemonClick.mock.calls.length).toBe(1);
  expect(mockHandleOnPokemonClick.mock.calls[0][0]).toBe('pikachu');

  const catchPokemon = tree.children[1];
  catchPokemon.props.onClick();
  expect(mockHandleOnCatchClick.mock.calls.length).toBe(1);
  expect(mockHandleOnCatchClick.mock.calls[0][0]).toBe('pikachu');
});

test('PokemonRow renders a "catch!" when pokemon is not my pokemon and style color red', () => {
  const component = renderer.create(
    <PokemonCard pokemon={pokemon} isMyPokemon={false} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('PokemonRow renders a "set free.." when pokemon is my pokemon', () => {
  const component = renderer.create(
    <PokemonCard pokemon={pokemon} isMyPokemon={true} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('PokemonRow getStyle returns empty object when isMyPokemon is false', () => {
  const pokemonRow = new PokemonCard();
  expect(pokemonRow.getStyle(false)).toEqual({});
});

test('PokemonRow getStyle returns style with color red when isMyPokemon is true', () => {
  const pokemonRow = new PokemonCard();
  expect(pokemonRow.getStyle(true)).toEqual({
    color: 'red',
  });
});
