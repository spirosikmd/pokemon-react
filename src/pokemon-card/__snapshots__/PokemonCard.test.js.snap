exports[`test PokemonRow renders a "catch!" when pokemon is not my pokemon and style color red 1`] = `
<div
  className="ui link card"
  href={undefined}
  onClick={[Function]}>
  <img
    alt={undefined}
    className="ui centered middle aligned image"
    height={100}
    onClick={[Function]}
    src={undefined}
    style={
      Object {
        "margin": "20px auto",
      }
    }
    width={undefined} />
  <div
    className="content"
    onClick={[Function]}>
    <div
      className="header">
      pikachu
    </div>
    <div
      className="description" />
  </div>
  <div
    className="extra content">
    <img
      alt={undefined}
      className="ui mini centered image"
      height={undefined}
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "borderRadius": "50%",
          "boxShadow": "0 1px 3px 0 #D4D4D5, 0 0 0 1px #D4D4D5",
        }
      }
      width={undefined} />
  </div>
</div>
`;

exports[`test PokemonRow renders a "set free.." when pokemon is my pokemon 1`] = `
<div
  className="ui link card"
  href={undefined}
  onClick={[Function]}>
  <img
    alt={undefined}
    className="ui centered middle aligned image"
    height={100}
    onClick={[Function]}
    src={undefined}
    style={
      Object {
        "margin": "20px auto",
      }
    }
    width={undefined} />
  <div
    className="content"
    onClick={[Function]}>
    <div
      className="header">
      pikachu
    </div>
    <div
      className="description" />
  </div>
  <div
    className="extra content">
    <img
      alt={undefined}
      className="ui mini centered image"
      height={undefined}
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "borderRadius": "50%",
          "boxShadow": "0 1px 3px 0 #D4D4D5, 0 0 0 1px #D4D4D5",
          "opacity": 0.3,
        }
      }
      width={undefined} />
  </div>
</div>
`;

exports[`test PokemonRow renders pokemon name catch button and handle data change 1`] = `
<div
  className="ui link card"
  href={undefined}
  onClick={[Function]}>
  <img
    alt={undefined}
    className="ui centered middle aligned image"
    height={100}
    onClick={[Function]}
    src={undefined}
    style={
      Object {
        "margin": "20px auto",
      }
    }
    width={undefined} />
  <div
    className="content"
    onClick={[Function]}>
    <div
      className="header">
      pikachu
    </div>
    <div
      className="description" />
  </div>
  <div
    className="extra content">
    <img
      alt={undefined}
      className="ui mini centered image"
      height={undefined}
      onClick={[Function]}
      src={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "borderRadius": "50%",
          "boxShadow": "0 1px 3px 0 #D4D4D5, 0 0 0 1px #D4D4D5",
        }
      }
      width={undefined} />
  </div>
</div>
`;
